source/common/stats/symbol_table_impl.cc

```diff
@@ -0,0 +1,86 @@
+#include "common/stats/symbol_table_impl.h"
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Stats {
+
+// TODO(ambuc): There is a possible performance optimization here for avoiding the encoding of IPs,
+// if they appear in stat names. We don't want to waste time symbolizing an integer as an integer,
+// if we can help it.
+StatNamePtr SymbolTableImpl::encode(const std::string& name) {
```

Comment:
I think encode() should take an absl::string_view as well.

Reply:
Agreed, implemented in c872073.